receiver/carbonreceiver/protocol/plaintext_parser.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package protocol
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+)
+
+// PlaintextParser converts a line of https://graphite.readthedocs.io/en/latest/feeding-carbon.html#the-plaintext-protocol,
+// treating tags per spec at https://graphite.readthedocs.io/en/latest/tags.html#carbon.
+type PlaintextParser struct{}
+
+var _ (Parser) = (*PlaintextParser)(nil)
+var _ (ParserConfig) = (*PlaintextParser)(nil)
+
+// BuildParser creates a new Parser instance that receives plaintext
+// Carbon data.
+func (p *PlaintextParser) BuildParser() (Parser, error) {
+	if p == nil {
+		return &PlaintextParser{}, nil
+	}
+	return p, nil
+}
+
+// Parse receives the string with plaintext data, aka line, in the Carbon
+// format and transforms it to the collector metric format. See
+// https://graphite.readthedocs.io/en/latest/feeding-carbon.html#the-plaintext-protocol.
+func (p PlaintextParser) Parse(line string) (*metricspb.Metric, error) {
+	parts := strings.SplitN(line, " ", 3)
```

Comment:
Would be nice to add a comment describing the format of `line`.

Reply:
Done